torchtext/csrc/sentencepiece.cpp

```diff
@@ -0,0 +1,89 @@
+#include <sentencepiece_trainer.h>
+#include <torch/script.h>
+
+namespace torchtext {
+namespace {
+
+struct SentencePiece : torch::CustomClassHolder {
+private:
+  sentencepiece::SentencePieceProcessor processor_;
+
+public:
+  std::string content_;
+  explicit SentencePiece(const std::string &content) : content_(content) {
+    const auto status = processor_.LoadFromSerializedProto(content_);
+    if (!status.ok()) {
+      throw std::runtime_error("Failed to load SentencePiece model. Error: " +
+                               status.ToString());
+    }
+  }
+
+  std::vector<std::string> Encode(const std::string &input) const {
+    std::vector<std::string> pieces;
+    processor_.Encode(input, &pieces);
+    return pieces;
+  }
+
+  std::vector<int64_t> EncodeAsIds(const std::string &input) const {
+    const auto val = processor_.EncodeAsIds(input);
+    return std::vector<int64_t>(val.begin(), val.end());
+  }
+
+  std::vector<std::string> EncodeAsPieces(const std::string &input) const {
+    return processor_.EncodeAsPieces(input);
+  }
+
+  int64_t GetPieceSize() const { return processor_.GetPieceSize(); }
+};
+
+// Registers our custom class with torch.
+static auto sentencepiece =
+    torch::class_<SentencePiece>("torchtext", "SentencePiece")
+        .def(torch::init<std::string>())
```

Comment:
At a minimum I would remove this unless necessary to prevent this being used in lieu of factory functions.